components/hab/src/main.rs

```diff
@@ -1651,23 +1652,23 @@ fn get_strategy_from_input(m: &ArgMatches) -> Option<UpdateStrategy> {
         .and_then(|f| UpdateStrategy::from_str(f).ok())
 }
 
-fn sup_addr_from_input(m: &ArgMatches) -> Result<SocketAddr> {
+fn sup_addr_from_input(m: &ArgMatches) -> Result<ListenCtlAddr> {
     match m.value_of("REMOTE_SUP") {
         Some(rs) => {
             let sup_addr = if rs.find(':').is_some() {
                 rs.to_string()
             } else {
-                format!("{}:{}", rs, protocol::ctl::DEFAULT_PORT)
+                format!("{}:{}", rs, ListenCtlAddr::default().port())
```

Comment:
I think we should pull all this logic into the `FromStr` implementation for `ListenCtlAddr`, since that is essentially what it is doing.

Going back to my previous suggestion of embedding the default port into the type itself... I think we can use that for this usecase to avoid creating a new value, just to grab the port from it.